Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -526,99 +538,68 @@ private void EnsureMixedRealityRequirements()
 
 #region MonoBehaviour Implementation
 
-        private static MixedRealityToolkit instance;
+        private static MixedRealityToolkit activeInstance;
         private static bool newInstanceBeingInitialized = false;
 
+        /// <summary>
+        /// Lock property for the Mixed Reality Toolkit to prevent reinitialization
+        /// </summary>
+        private static readonly object initializedLock = new object();
+
         /// <summary>
         /// Returns the Singleton instance of the classes type.
         /// If no instance is found, then we search for an instance in the scene.
-        /// If more than one instance is found, we log an error and no instance is returned.
+        /// If more than one instance is found, only the first instance is activated. All others are deactivated.
         /// </summary>
         public static MixedRealityToolkit Instance
         {
             get
             {
-                if (IsInitialized)
+                if (activeInstance != null)
                 {
-                    return instance;
-                }
-
-                if (Application.isPlaying && !searchForInstance)
-                {
-                    return null;
+                    return activeInstance;
                 }
 
-
                 var objects = FindObjectsOfType<MixedRealityToolkit>();
-                searchForInstance = false;
-                MixedRealityToolkit newInstance;
 
                 switch (objects.Length)
                 {
                     case 0:
-#if UNITY_EDITOR
-                        // Generating a new instance during app shutdown in the editor allows instances to persist into edit mode, which may cause unwanted behavior
-                        if (isApplicationQuitting)
-                        {
-                            Debug.LogWarning("Trying to find instance during application shutdown.");
-                            return null;
-                        }
-#endif
-                        Debug.Assert(!newInstanceBeingInitialized, "We shouldn't be initializing another MixedRealityToolkit unless we errored on the previous.");
-                        newInstanceBeingInitialized = true;
-                        newInstance = new GameObject(nameof(MixedRealityToolkit)).AddComponent<MixedRealityToolkit>();
                         break;
+
                     case 1:
-                        newInstanceBeingInitialized = false;
-                        newInstance = objects[0];
+                        RegisterInstance(objects[0]);
```

Comment:
I don't see the need to register the instance in the getter. When the instance is created, it should register itself then, not when someone calls it.